include/revng/ADT/STLExtras.h

```diff
@@ -286,10 +286,9 @@ inline llvm::ArrayRef<uint8_t> toArrayRef(llvm::StringRef Data) {
 // append
 //
 template<std::ranges::sized_range FromType, std::ranges::sized_range ToType>
-auto append(FromType &&From, ToType &To) {
-  size_t ExistingElementCount = To.size();
-  To.resize(ExistingElementCount + From.size());
-  return llvm::copy(From, std::next(To.begin(), ExistingElementCount));
+void append(FromType &&From, ToType &To) {
+  To.reserve(To.size() + From.size());
+  llvm::copy(From, std::back_inserter(To));
 }
```

Comment:
@Cvelth do you approve?

Reply:
If we go this route, it's probably a good idea to actually enforce that fact that `To` must be a vector-like: https://godbolt.org/z/MnxnYY4vc

I guess the old approach also assumed that (since it used `resize`). So my bad on not constraining the interface correctly.

But, while on topic, let's make it right (tm).
Here's what I'd suggest: https://godbolt.org/z/neYsYbjoq

What do you, guys, think?

Reply:
We could also do something cool like
```cpp
template <typename T>
concept HasReserve = requires(T &&V) { { V.reserve(5) }; };
// . . .
  if constexpr(HasReserve<ToType>)
    To.reserve(To.size() + From.size());
```

But even I think that might be an overkill.

Reply:
Wouldn't `std::ranges::move` imply that `From` is always an rvalue?
Wouldn't this be better?

```cpp
template<std::ranges::sized_range FromType, std::ranges::sized_range ToType>
auto append(FromType &&From, ToType &To) {
  if constexpr (std::is_lvalue_reference_v<FromType>)
    return std::ranges::copy(From, std::inserter(To, To.end()));
  else
    return std::ranges::move(From, std::inserter(To, To.end()));
}
```

Reply:
Wait, yeah, I wasn't paying attention, thanks for catching that.

Reply:
I've done some testing and using `reserve` cuts times considerably, so I've added the version above with `HasReserve`